Firmware/ultralcd.cpp

```diff
@@ -5297,81 +5314,96 @@ static void lcd_main_menu()
 #endif
 
     // only allow starting SD print if hardware errors (temperature or fan) are cleared
-    if(!get_temp_error()
-            && !printer_active()
+    if (!printer_recovering()
+        && !printer_active()
+        && !get_temp_error()
+        && !printer_active()
 #ifdef FANCHECK
-            && fan_check_error != EFCE_REPORTED
+        && fan_check_error != EFCE_REPORTED
 #endif //FANCHECK
-    )
-    {
+        ) {
 #ifdef SDSUPPORT //!@todo SDSUPPORT undefined creates several issues in source code
-        if (card.mounted || lcd_commands_type != LcdCommands::Idle) {
-            if (!card.isFileOpen()) {
-                if (!usb_timer.running() && (lcd_commands_type == LcdCommands::Idle)) {
-                    bMain=true;               // flag ('fake parameter') for 'lcd_sdcard_menu()' function
-                    MENU_ITEM_SUBMENU_P(_T(MSG_CARD_MENU), lcd_sdcard_menu);
-                }
+            if (card.mounted
+                || lcd_commands_type != LcdCommands::Idle) {
+                if (!card.isFileOpen()) {
+                    if (!usb_timer.running()
+                        && (lcd_commands_type == LcdCommands::Idle)) {
+                        bMain=true;               // flag ('fake parameter') for 'lcd_sdcard_menu()' function
+                        MENU_ITEM_SUBMENU_P(_T(MSG_CARD_MENU), lcd_sdcard_menu);
+                    }
 #if SDCARDDETECT < 1
-            MENU_ITEM_GCODE_P(_i("Change SD card"), PSTR("M21"));  // SD-card changed by user ////MSG_CNG_SDCARD c=18
+             MENU_ITEM_GCODE_P(_i("Change SD card"), PSTR("M21"));  // SD-card changed by user ////MSG_CNG_SDCARD c=18
 #endif //SDCARDDETECT
-            }
-        } else {
-            bMain=true;                                   // flag (i.e. 'fake parameter') for 'lcd_sdcard_menu()' function
-            MENU_ITEM_SUBMENU_P(_i("No SD card"), lcd_sdcard_menu); ////MSG_NO_CARD c=18
+                }
+            } else {
+                bMain=true;                                   // flag (i.e. 'fake parameter') for 'lcd_sdcard_menu()' function
+                MENU_ITEM_BACK_P(_i("No SD card")); ////MSG_NO_CARD c=18
 #if SDCARDDETECT < 1
             MENU_ITEM_GCODE_P(_i("Init. SD card"), PSTR("M21")); // Manually initialize the SD-card via user interface ////MSG_INIT_SDCARD c=18
 #endif //SDCARDDETECT
-        }
+         }
 #endif //SDSUPPORT
-    }
-
-    if(!printer_active() && !farm_mode) {
-        const int8_t sheet = eeprom_read_byte(&(EEPROM_Sheets_base->active_sheet));
-        const int8_t nextSheet = eeprom_next_initialized_sheet(sheet);
-        if ((nextSheet >= 0) && (sheet != nextSheet)) { // show menu only if we have 2 or more sheets initialized
-            MENU_ITEM_FUNCTION_E(EEPROM_Sheets_base->s[sheet], eeprom_switch_to_next_sheet);
-        }
+        if(!printer_active()
+            && !farm_mode) {
```

Comment:
Why is this split into two lines? Can easily fit one line and is more readable IMO